### ISSUE TITLE
New rule: no-submodule-imports

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#3051]
I'd like a new rule to disallow reaching into submodules in import paths. Similar to [eslint-plugin-import/no-internal-modules](https://github.com/benmosher/eslint-plugin-import/blob/ccd93942a0b9f59f1b698672ba167ece1aa9f5f1/docs/rules/no-internal-modules.md) (but we can't call it that since TypeScript has its own notion of "internal module").

```ts
// bad
import { Keys } from "@blueprintjs/core/dist/common";

// good
import { Keys } from "@blueprintjs/core";
```

with rule config

```json
"no-submodule-imports": {
  "options": [
    "@blueprinjs/core"
  ]
}
```

**Comment by aervin** [comment]
Will this be very similar to the "import-blacklist" rule? If so, I think I could give it a try.

**Comment by ajafff** [comment]
Yes, this rule will be very similar to import-blacklist. In fact there will
only be a different condition to show an error. The rest of the logic can
be exactly the same.

Am 02.08.2017 18:07 schrieb "aervin_" <notifications@github.com>:

Will this be very similar to the "import-blacklist" rule? If so, I think I
could give it a try.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/palantir/tslint/issues/3051#issuecomment-319719610>,
or mute
the thread
